Scripts/static-analyzer/Sources/Analyzer/SourceNavigator.swift

```diff
@@ -0,0 +1,160 @@
+    /// three expressions. The expressions are (the dash lines in the example are each expression's "byte range"):
+    /// 1. The `map` function: `([Int]) -> ((Int) throws -> String) throws -> [String]`.
+    /// 2. The `Array.map` function: `((Int) throws -> String) throws -> [String]`.
+    /// 3. The return type of the whole `map` function call: `[String]`.
+    ///
+    /// Note that all of the above are specialized signature, rather than the declared generic type.
+    public func expressions(at location: SymbolLocation) throws -> [Expression] {
+        let compilerArguments = compilerInvocations.compilerArguments(forFileAt: location.path)
+        let targetByteOffset = try Int64(StringView(String(contentsOfFile: location.path)).byteOffset(forLine: Int64(location.line), bytePosition: Int64(location.utf8Column))!.value)
+
+        let request = Request.expressionType(file: location.path, compilerArguments: compilerArguments)
+        let response = try request.send()
+
+        guard let expressionTypeList = response["key.expression_type_list"] as? [[String: SourceKitRepresentable]] else {
+            throw SourceKitResponseError.unexpectedType(key: "key.expression_type_list")
+        }
+
+        return try expressionTypeList
+            .map { try Expression(sourceKitExpressionTypeResponse: $0) }
+            .filter { $0.byteRange.contains(targetByteOffset) }
+            .sorted { $0.byteRange.count < $1.byteRange.count }
+    }
+
+    public func isInheritence(subclass subclassName: String, superclass superclassName: String, usedAt location: SymbolLocation? = nil) throws -> Bool {
+        do {
+            let subclass = try resolveType(named: subclassName)
+            let superclass = try resolveType(named: superclassName)
+            return superclasses(of: subclass).map { $0.usr }.contains(superclass.usr)
+        } catch {
+            print(error)
+        }
+
+        if let location {
+            print("Try SourceKit")
+            return try isInheritence(subclass: subclassName, superclass: superclassName, usedAt: location, compilerArguments: compilerInvocations.compilerArguments(forFileAt: location.path))
+        }
+
+        return false
+    }
+
+    private func isInheritence(subclass typename: String, moduleName: String? = nil, superclass: String, usedAt location : SymbolLocation, compilerArguments: [String]) throws -> Bool {
+        let fileManager = FileManager.default
+        let tempFile = fileManager.temporaryDirectory.appendingPathComponent("temp.swift")
+        var tempFileContent = try String(contentsOfFile: location.path)
+
+        if let moduleName, moduleName.starts(with: "_") == false, moduleName != appModuleName {
+            tempFileContent = tempFileContent.inserting("import \(moduleName)", atLine: 1)
+        }
+
+
+        let typeCheckCode = "let __injected_variable: \(typename)? = nil"
```

Comment:
https://github.com/wordpress-mobile/WordPress-iOS/pull/21212 had ane explanation for this clever workaround. Have you considered adding it here, in the code, too?